go/db/tls.go

```diff
@@ -90,53 +87,66 @@ func requiresTLS(host string, port int, mysql_uri string) bool {
 	return required
 }
 
-// Create a TLS configuration from the config supplied CA, Certificate, and Private key.
-// Register the TLS config with the mysql drivers as the "topology" config
-// Modify the supplied URI to call the TLS config
 func SetupMySQLTopologyTLS(uri string) (string, error) {
-	if !topologyTLSConfigured {
-		tlsConfig, err := ssl.NewTLSConfig(config.Config.MySQLTopologySSLCAFile, !config.Config.MySQLTopologySSLSkipVerify)
-		// Drop to TLS 1.0 for talking to MySQL
-		tlsConfig.MinVersion = tls.VersionTLS10
-		if err != nil {
-			return "", log.Errorf("Can't create TLS configuration for Topology connection %s: %s", uri, err)
-		}
-		tlsConfig.InsecureSkipVerify = config.Config.MySQLTopologySSLSkipVerify
+	c := connectionConfig{
+		uri:              uri,
+		caCertFile:       config.Config.MySQLTopologySSLCAFile,
+		useClientAuth:    config.Config.MySQLTopologyUseMutualTLS,
+		clientCertFile:   config.Config.MySQLTopologySSLCertFile,
+		clientKeyFile:    config.Config.MySQLTopologySSLPrivateKeyFile,
+		verifyClientCert: !config.Config.MySQLTopologySSLSkipVerify,
+		verifyServerCert: !config.Config.MySQLTopologySSLSkipVerify,
+	}
+	return setupTls("topology", c)
+}
 
-		if config.Config.MySQLTopologyUseMutualTLS ||
-			config.Config.MySQLTopologySSLCertFile != "" ||
-			config.Config.MySQLTopologySSLPrivateKeyFile != "" {
-			if err = ssl.AppendKeyPair(tlsConfig, config.Config.MySQLTopologySSLCertFile, config.Config.MySQLTopologySSLPrivateKeyFile); err != nil {
-				return "", log.Errorf("Can't setup TLS key pairs for %s: %s", uri, err)
-			}
-		}
-		if err = mysql.RegisterTLSConfig("topology", tlsConfig); err != nil {
-			return "", log.Errorf("Can't register mysql TLS config for topology: %s", err)
-		}
-		topologyTLSConfigured = true
+func SetupMySQLOrchestratorTLS(uri string) (string, error) {
+	c := connectionConfig{
+		uri:              uri,
+		caCertFile:       config.Config.MySQLOrchestratorSSLCAFile,
+		useClientAuth:    config.Config.MySQLOrchestratorUseMutualTLS,
+		clientCertFile:   config.Config.MySQLOrchestratorSSLCertFile,
+		clientKeyFile:    config.Config.MySQLOrchestratorSSLPrivateKeyFile,
+		verifyClientCert: true, // FIXME: Should this have been SSLSkipVerify?
```

Comment:
This is to keep existing behaviour. I wasn't completely certain if this is a bug, as it's not consistent with `SetupMySQLTopologyTLS`. In this case, I left the original value.